ddl/rollingback.go

```diff
@@ -106,6 +106,32 @@ func convertNotStartAddIdxJob2RollbackJob(t *meta.Meta, job *model.Job, occuredE
 	return convertAddIdxJob2RollbackJob(t, job, tblInfo, indexInfo, occuredErr)
 }
 
+// rollingbackModifyColumn change the modifying-column job into rolling back state.
+// Since modifying column job has two types: normal-type and reorg-type, we should handle it respectively.
+// normal-type has only two states:    None -> Public
+// reorg-type has five states:         None -> Delete-only -> Write-only -> Write-org -> Public
+func rollingbackModifyColumn(t *meta.Meta, job *model.Job) (ver int64, err error) {
+	_, _, oldCol, jp, err := getModifyColumnInfo(t, job)
+	if err != nil {
+		return ver, err
+	}
+	if !needChangeColumnData(oldCol, jp.newCol) {
+		// normal-type rolling back
+		// TODO: when change null to not null, although state is unchanged with none, the oldCol flag's has been changed to preNullInsertFlag.
+		// TODO: To roll back this kind of normal job, it is necessary to mark the state as JobStateRollingback to restore the old col's flag.
```

Comment:
Why not handle it？It seems that we can handle it like https://github.com/pingcap/tidb/pull/19531/files#diff-0ef842ac66131b78e2b1b259c81fc5f4R700.

Reply:
Em~ good catch! addressed